joomla/joomla-cms
# [4.0] Implementing OpenSearch in com_finder

This implements the OpenSearch specification into com_finder. OpenSearch allows you to add a sites search to the search engines of your browser for example. Funnily, this code was partially in com_finder already, but has been disabled. This PR makes that working again.

### How to test
* Open your test website with Firefox and go to a page that has either the mod_finder module displayed or the Smart Search search form.
* In your address bar, there is a menu (the 3 dots). See that there is no "Add Search menu" option (or something similar. It is translated to german in my FF, so I don't know what the original english description is.) or that there is just an entry for com_search. (Click on that and Firefox should shortly after throw an error message.)
* Apply this PR
* Now see that a new entry has shown up, named "Search {Site Title}". Click on that and shortly after see a small success notification.
* Go to your search settings in Firefox and set to display an extra search box and to use suggestions.
* In the search box, type in a fraction of a word, like "joom" for Joomla. See suggestions being displayed. Select a suggestion and click on search.
* See that you are being directed to your sites search page.

Note: OpenSearch doesn't work in 3.x for com_search right now either...
